lib/src/chart/scatter_chart/scatter_chart_painter.dart

```diff
@@ -290,6 +290,67 @@ class ScatterChartPainter extends AxisChartPainter<ScatterChartData> {
       );
     }
 
+    if (data.scatterLabelSettings.showLabel) {
+      for (final scatterSpot in data.scatterSpots) {
+        int spotIndex = data.scatterSpots.indexOf(scatterSpot);
```

Comment:
`indexOf()` call takes O(N) to find the index. While we can use an indexed `for` to have both `index` and `scatterSpot` together.

like:
```dart
for (int i = 0; i < data.scatterSpots.length; i++) {
   scatterSpot = data.scatterSpots[i]
   int spotIndex = i;
}
```